src/server/accounting.c

```diff
@@ -2274,40 +2284,25 @@ void log_alter_records_for_attrs(job *pjob, svrattrl *plist) {
  * @returns void 
  */
 void
-log_suspend_resume_record(job *pjob, char *user, char *host, char *signal_type)
+log_suspend_resume_record(job *pjob, char *user, char *host, int acct_type)
 {
-	char *log_record = NULL;
+	char *buf = NULL;
 	char *tmp;
-	struct svrattrl *patlist = NULL;
 	char *resc_used;
 	int resc_used_size = 0;
-	pbs_list_head temp_head;
-	int acct_type = PBS_ACCT_RESUME;
 	int rc;
 
-	rc = pbs_asprintf(&log_record, "requestor=%s@%s", user, host);
+	rc = pbs_asprintf(&buf, "requestor=%s@%s", user, host);
 
-	if (strcmp(signal_type, SIG_SUSPEND)==0 || strcmp(signal_type, SIG_ADMIN_SUSPEND) == 0) {
-		acct_type = PBS_ACCT_SUSPEND;
+	if (acct_type == PBS_ACCT_SUSPEND) {
 		
 		if (pjob->ji_wattr[JOB_ATR_resc_released].at_flags & ATR_VFLAG_SET) {
-			tmp = log_record;
-			rc = pbs_asprintf(&log_record, "%s resources_released=%s", tmp, 
+			tmp = buf;
+			rc = pbs_asprintf(&buf, "%s resources_released=%s", tmp,
```

Comment:
Don't use tmp and buf in the same call to pbs_asprintf().  Use snprintf() to write into a local buffer and then pbs_strcat() to concatenate it onto the final string.

Reply:
I like the idea of using pbs_strcat as well. It might also help reduce the number of mallocs, each call of pbs_asprintf() does an unconditional malloc, whereas pbs_strcat will do it only when the current buffer runs out.